unlock-app/src/components/interface/checkout/alpha/Checkout/Minting.tsx

```diff
@@ -50,6 +53,36 @@ export function Minting({ injectedProvider, send, onClose, state }: Props) {
   const { mint, lock } = state.context
   const processing = mint?.status === 'PROCESSING'
 
+  useEffect(() => {
+    async function waitForConfirmation() {
+      try {
+        if (!mint?.transactionHash && mint?.status !== 'PROCESSING') {
+          return
+        }
+        const network = config.networks[lock!.network]
+        if (network) {
+          const provider = new ethers.providers.JsonRpcProvider(
```

Comment:
I wonder if this hook could be moved to its own file as this is a pattern we will re-use?

Reply:
IMO, not necessary. Unless this exact code is copy pasted a few times, extracting a hook creates unnecessary abstraction.